text/0000-safe-uninit-bytes.md

```diff
@@ -0,0 +1,81 @@
+- Feature Name: safe_uninit_bytes
+- Start Date: 2015-07-21
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+Define all exclusively owned, allocated, and never-initialized memory as being
+one-way re-interpretable as valid `[u8]`. In other words, all (allocated) byte
+arrays are valid byte arrays.
+
+# Motivation
+
+This would make it possible to safely define (i.e. the behavior would both be
+safe and defined):
+
+```rust
+impl Vec<u8> {
+    pub fn grow_uninitialized(&mut self, amount: usize) {
+        self.reserve(amount);
+        let len = self.len();
+        self.set_len(len + amount);
+    }
+}
+```
+
+And would make it much easier to implement fast IO code.
+
+# Design/Argument
+
+Every addressable byte in allocated memory is a valid u8 (byte) by definition.
+On Linux at least, one can read `/proc/self/mem` into a buffer so the following
+two functions are (virtually) indistinguishable at runtime (on Linux):
```

Comment:
Except for the fact that `/proc` is not necessarily mounted, e.g. a strict chroot might opt out of it.